uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/NewFuseProjectVerifyBTNTest.java

```diff
@@ -0,0 +1,177 @@
+import org.jboss.tools.fuse.reddeer.condition.LabelContainsText;
+import org.jboss.tools.fuse.reddeer.perspectives.FuseIntegrationPerspective;
+import org.jboss.tools.fuse.reddeer.preference.ConsolePreferenceUtil;
+import org.jboss.tools.fuse.reddeer.wizard.NewFuseIntegrationProjectWizard;
+import org.jboss.tools.fuse.ui.bot.tests.utils.LogChecker;
+import org.jboss.tools.fuse.ui.bot.tests.utils.ProjectFactory;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Verifies functionality of 'Verify' button on New Fuse Integration Project wizard
+ * 
+ * @author djelinek
+ */
+@CleanWorkspace
+@CleanErrorLog
+@OpenPerspective(FuseIntegrationPerspective.class)
+@RunWith(RedDeerSuite.class)
+public class NewFuseProjectVerifyBTNTest {
+
+	protected Logger log = Logger.getLogger(NewFuseProjectVerifyBTNTest.class);
+	
+	private static final String PROJECT_NAME = "test";
+	
+	private static final String EXISTING_CAMEL_VERSION = SupportedCamelVersions.CAMEL_2_17_0_REDHAT_630187;
+	
+	private static final String NON_EXISTING_CAMEL_VERSION = "0.0.0";
+	
+	private static final String INTERRUPT_CAMEL_VERSION = "0.0.1";
+	
+	private static final String COMMUNITY_CAMEL_VERSION = "2.20.2";
+	
+	private NewFuseIntegrationProjectWizard wizard;
+	
+	@BeforeClass
+	public static void initSetup() {
+		new WorkbenchShell().maximize();
+		ConsolePreferenceUtil.setConsoleOpenOnError(false);
+		ConsolePreferenceUtil.setConsoleOpenOnOutput(false);
+	}
+	
+	@Before
+	public void prepareTestEnvironment() {
+		wizard = new NewFuseIntegrationProjectWizard();
+		wizard.open();
+		wizard.setProjectName(PROJECT_NAME);
+		wizard.next();
+	} 
+	
+	@After
+	public void cleanSetup() {
+		LogView view = new LogView(); 
+		view.open();
+		view.deleteLog();
+		WorkbenchShellHandler.getInstance().closeAllNonWorbenchShells();
+		EditorHandler.getInstance().closeAll(true);
+		ProjectFactory.deleteAllProjects();
+	}
+	
+	/**
+	 * Test tries to verify existing supported Camel version
+	 */
+	@Test
+	public void testExistingCamelVersion() {	
+		wizard.setCamelVersion(EXISTING_CAMEL_VERSION);
+		wizard.verifyBTN();
+		
+		waiting();
+		finish();
+		LogChecker.assertNoFuseError();
+	}
+	
+	/**
+	 * Test tries to verify non-existing Camel version
+	 */
+	@Test
+	public void testNonExistingCamelVersion() {
+		wizard.setCamelVersion(NON_EXISTING_CAMEL_VERSION);
+		wizard.verifyBTN();
+		
+		waiting();
+		assertButtons();
+		cancel();
+		LogChecker.assertNoFuseError();
+	}
+	
+	/**
+	 * Test tries to verify community Camel version
+	 */
+	@Test
+	public void testCommunityCamelVersion() {
+		wizard.setCamelVersion(COMMUNITY_CAMEL_VERSION);
+		wizard.verifyBTN();
+		
+		waiting();
+		finish();
+		LogChecker.assertNoFuseError();
+	}
+	
+	/**
+	 * Test tries to interrupt verifying of project Camel version 
+	 */
+	@Test
+	public void testInterruptVerifying() {
```

Comment:
This test method ends with error every time
```Exception in test: testInterruptVerifying no-configuration(org.jboss.tools.fuse.ui.bot.tests.NewFuseProjectVerifyBTNTest)
org.eclipse.reddeer.core.exception.CoreLayerException: No matching widget found with Matcher matching when all matchers match: [Matcher matching widget with the same type as or type extending class org.eclipse.swt.widgets.Label, Matcher matching widget which text matches: "Checking availability for Camel version 0.0.1. Please wait..."]
...
at org.jboss.tools.fuse.ui.bot.tests.NewFuseProjectVerifyBTNTest.waiting(NewFuseProjectVerifyBTNTest.java:175)
	at org.jboss.tools.fuse.ui.bot.tests.NewFuseProjectVerifyBTNTest.testInterruptVerifying(NewFuseProjectVerifyBTNTest.java:145)
...
```
Maybe the _WaitWhile_, in waiting() method, could not throw the exception, when the condition is not met.

Reply:
There is used WaitWhile condition, but I made some changes and it is ready for review now